### ISSUE TITLE
Too much time waste on Labels::default

### ISSUE THREAD

**Issue by taikulawo** [metrics-rs/metrics#547]
We found metrics-tracing-context use too much time on CPU cycle. 
The callstack show `Labels::default` are in busyloop.
https://github.com/metrics-rs/metrics/blob/e6cf12457b4e9b81eb319f4aab70a573be7998b3/metrics-tracing-context/src/tracing_integration.rs#L52

It maybe related https://github.com/EVaillant/lockfree-object-pool/issues/5#issuecomment-2550462336

**Comment by tobz** [comment]
Yeah, `lockfree-object-pool` is likely not the best possible object pool we could be using here. I'm open to PRs for changing/improving things if you have a workload you can use to demonstrate the improvements to switching to something else.
